sources/newRif/rif_utils.py

```diff
@@ -1,33 +1,91 @@
 import os
 import csv
 import re
+import django
+django.setup()
 from sefaria.model import *
 
 path = os.path.dirname(os.path.abspath("top_level_file.txt"))
 
+def cleanspaces(string):
+    while any(space in string for space in ['  ', '( ', ' )', ' :', ' .']):
+        for space in ['  ', '( ', ' )', ' :', ' .']:
+            string = string.replace(space, space.replace(' ', '', 1))
+    return string.strip()

```

Comment:
Let's discuss this method. My intuition is that this is a lot more expensive than the operation needs to be. But I'm not sure I understand the problem being addressed.